lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -85,19 +85,26 @@ def resource_name_for(env, status)
         end
 
         def set_request_tags!(request_span, env, status, headers, response, original_env)
+          # http://www.rubydoc.info/github/rack/rack/file/SPEC
+          # The source of truth in Rack is the PATH_INFO key that holds the
+          # URL for the current request; but some frameworks may override that
+          # value, especially during exception handling.
+          #
+          # Because of this, we prefer to use REQUEST_URI, if available, which is the
+          # relative path + query string, and doesn't mutate.
+          #
+          # REQUEST_URI is only available depending on what web server is running though.
+          # So when its not available, we want the original, unmutated PATH_INFO, which
+          # is just the relative path without query strings.
+          url = env['REQUEST_URI'] || original_env['PATH_INFO']
```

Comment:
let's keep the behavior as before, we'll re-evaluate keeping or not the query string later